ee/WFLY-19776_Jakarta_Data_in_standard_WildFly.adoc

```diff
@@ -0,0 +1,132 @@
+---
+categories:
+ - ee
+stability-level: preview
+issue: https://github.com/wildfly/wildfly-proposals/issues/608
+feature-team:
+ developer: bstansberry
+ sme:
+  - yrodiere
+ outside-perspective:
+  - jasondlee
+# If this issue tracks the promotion to a higher stability level of a previously
+# completed feature, provide the URL of the https://github.com/wildfly/wildfly-proposals/issues
+# issue that was used to track the previous feature.
+promotes:
+# This should be blank during initial development of a feature. It may be used
+# after the feature is completed if a subsequent issue is field to track promotion
+# of this feature to a higher stability level
+promoted-by:
+---
+= <INSERT TITLE HERE>
+:author:            Brian Stansberry
+:email:             bstansbe@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+
+== Overview
+
+In WildFly 34 we link:https://issues.redhat.com/browse/WFLY-19397[introduced] `preview` stability support for link:https://jakarta.ee/specifications/data/1.0/jakarta-data-1.0[Jakarta Data] into WildFly Preview. Here we propose making it available in standard WildFly, also at `preview` stability. This will allow a larger part of our user base to use the technology.
+
+While Jakarta Data will first appear in Jakarta EE as part of EE 11, its API is not bound to the EE 11 variants of other EE APIs, and our implementation, Hibernate Data Repositories, is part of Hibernate ORM 6.6, which implements the EE 10 Jakarta Persistence 3.1 specification. I've discussed with the ORM team whether this should work in an EE 10 server and they confirmed that it will and are in favor of making it more widely available to WildFly users.
+
+=== User Stories
+
+I'm an application developer and I'd like to be less concerned with the mechanics of manipulating persistent data. I'd like to define my entities and then write an interface that describes the CRUD and data access operations I need. I code against that interface and let WildFly take care of generating the interface implementation that deals with the DB.
+
+I'm an application developer and am familiar with the Repository pattern from use of Spring Data. I'd like to use that pattern in WildFly in non-Spring applications.
+
+I'm an application developer who's used Deltaspike Data, which uses a repository pattern. But it doesn't support EE > 8, so I'm looking for a similar technology to run on the latest WildFly releases.
```

Comment:
You might want to mention that users need to be fine with having less flexibility, in particular:

* Simpler query language.
* No lazy loading -- or only explicit lazy loading by using Hibernate ORM native APIs.
* More limited support for dynamic queries; I think there is some support for dynamic predicates, but it far from the Criteria API.
* And so on.

Which is fine, but it does mean that it's not just about what developers "like", it's also about the complexity of their use case: Jakarta Data is great in many cases, but may be too limited in others.

Note that it's perfectly possible to use standard JPA, native Hibernate ORM APIs, and Jakarta Data in the same application. Depending on how the stateless session is injected into Jakarta Data, it's probably even possible to use Hibernate ORM APIs (`StatelessSession`) and Jakarta Data in the same transaction.

Reply:
Thanks @yrodiere. I incorporated a lot of your info above in the first user story.